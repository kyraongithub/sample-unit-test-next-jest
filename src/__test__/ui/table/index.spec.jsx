import { render } from '@testing-library/react';
import Table from '@/components/ui/table';

describe('Table Component', () => {
  it('should render table with no data', () => {
    const { container } = render(<Table />);
    expect(container).toMatchSnapshot();
  });

  it('should render table with data', () => {
    const TABLE_HEAD = ['Name', 'Email'];
    const TABLE_DATA = [
      {
        name: 'John',
        email: '7XQp0@example.com',
      },
      {
        name: 'Christ',
        email: 'abcd@example.com',
      },
    ];
    const { container } = render(<Table head={TABLE_HEAD} data={TABLE_DATA} />);
    expect(container).toMatchSnapshot();
  });

  it('should render table with data validate row count', () => {
    const TABLE_HEAD = ['Name', 'Email'];
    const TABLE_DATA = [
      {
        name: 'John',
        email: '7XQp0@example.com',
      },
      {
        name: 'Christ',
        email: 'abcd@example.com',
      },
    ];
    const { getAllByRole } = render(
      <Table head={TABLE_HEAD} data={TABLE_DATA} />
    );
    const tableRows = getAllByRole('row');
    expect(tableRows).toHaveLength(3);
  });

  it('should render table with data validate cell count', () => {
    const TABLE_HEAD = ['Name', 'Email'];
    const TABLE_DATA = [
      {
        name: 'John',
        email: '7XQp0@example.com',
      },
      {
        name: 'Christ',
        email: 'abcd@example.com',
      },
    ];
    const { getAllByRole } = render(
      <Table head={TABLE_HEAD} data={TABLE_DATA} />
    );
    const tableRows = getAllByRole('cell');
    expect(tableRows).toHaveLength(4);
  });
});
