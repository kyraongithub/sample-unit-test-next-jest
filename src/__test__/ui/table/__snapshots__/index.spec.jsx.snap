// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component should render table with data 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Email
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          John
        </td>
        <td>
          7XQp0@example.com
        </td>
      </tr>
      <tr>
        <td>
          Christ
        </td>
        <td>
          abcd@example.com
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table Component should render table with no data 1`] = `
<div>
  <table>
    <thead>
      <tr />
    </thead>
    <tbody />
  </table>
</div>
`;
